waku/v2/node/start_network2.nim

```diff
@@ -68,19 +90,68 @@ proc initNodeCmd(shift: int, staticNodes: seq[string] = @[], master = false, lab
   if staticNodes.len > 0:
     for staticNode in staticNodes:
       result.cmd &= "--staticnode:" & staticNode & " "
+
+  # Scenarios using discv5
+  result.cmd &= "--nat:extip:" & discv5ExtIpAddr & " "
+  if discv5BootStrapEnrs.len > 0:
+    for enr in discv5BootStrapEnrs:
+      result.cmd &= "--discv5-bootstrap-node:" & enr & " "
+
   result.shift = shift
   result.label = label
   result.master = master
   result.address = listenStr
 
+  # Scenarios using discv5
+  # We have to manually build the ENR which the node ran with the currently build node command will generate,
+  # because we need to know it before the node starts in order to be able to provide it as bootstrap information.
+  # Note: the ENR built here is not exactly the same as we don't integrate the waku2 key in this test scenario.
+  let enr = enr.Record.init(1,
+    keys.PrivateKey.fromHex(hkey).expect("could not convert priv key from hex"),
+    some(stewNet.ValidIpAddress.init(discv5ExtIpAddr)),
+    some(Port(discv5Port)), # tcp-port
+    some(Port(discv5Port))) # udp-port
+    .expect("Record within size limits")
+  result.enrUri = "enr:" & enr.toBase64
+
   info "Node command created.", cmd=result.cmd, address = result.address
 
+
+# # Scenarios without discv5
+# proc starNetwork(amount: int): seq[NodeInfo] =
```

Comment:
Is there a reason to comment this out? As opposed to defining a new function with other name like, `discv5Network` and have it be listed as a topology here https://github.com/status-im/nim-waku/blob/7168bf6606938fff97435251238aece802606793/waku/v2/node/start_network2.nim#L232-L238

I don't quite understand `bootstrapFleetNetwork` vs the changes in `starNetwork`. How come they aren't a single topology discv5Something?

Reply:
These topologies are ment to be independent of the scenarios.
`Star` consists of a single `master node` which all other nodes use as bootstrap node.
`BootstrapFleet` consists of a fully connected fleet (whose size can be set via a parameter). All nodes get the set of fleet nodes as bootstrap nodes. `BootstrapFleet` can be seen as a generalization of `Star`.

I used `BootstrapFleet` as the topology for the discv5 tests to mimic the real world fleet setup.

> Is there a reason to comment this out? 

There is one default `Star` setup and one for discv5.
Because `BootstrapFleet` is new, and I just focused on the discv5 setup, I did not add a default version for it.
I could add one, if desired.

Reply:
(resolved in 1:1 meeting)